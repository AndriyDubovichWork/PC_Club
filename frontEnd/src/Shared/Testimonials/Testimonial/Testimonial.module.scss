.Text {
  font-weight: 500;
  font-size: clamp(0.6rem, 1vw, 2rem);
}
.Author {
  margin-top: 5vh;

  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;

  .Img {
    width: 5vw;
    border-radius: 50%;
  }
}
