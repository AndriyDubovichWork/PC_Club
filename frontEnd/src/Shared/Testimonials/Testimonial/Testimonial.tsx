import React from 'react';
import style from './Testimonial.module.scss';
import { Paper } from '@mui/material';
import FormatQuoteIcon from '@mui/icons-material/FormatQuote';

type TestimonialPropsT = {
  text: string;
  name: string;
  img: string;
};

const Testimonial = ({ img, text, name }: TestimonialPropsT) => {
  return (
    <Paper sx={{ padding: '10vh 2vw' }}>
      <div>
        <span className={style.Text}>"{text}"</span>
      </div>
      <div className={style.Author}>
        <img loading='lazy' src={img} alt='testimonial' className={style.Img} />
        <h3>{name}</h3>
        <FormatQuoteIcon sx={{ fontSize: '5rem', color: '#ff4d30' }} />
      </div>
    </Paper>
  );
};

export default Testimonial;
