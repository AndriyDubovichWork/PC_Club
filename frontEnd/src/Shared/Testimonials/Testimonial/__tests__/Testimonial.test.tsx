import { render, screen, cleanup } from '@testing-library/react';
import Testimonial from '../Testimonial';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Testimonial match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route
            path='/PC_Club/'
            element={
              <Testimonial
                name='Harry Potter'
                text='I had a great experience at PC Club! The staff was friendly and helpful. I highly recommend this store.'
                img={require('../../assets/harry.jpg')}
              />
            }
          />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Testimonial contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route
          path='/PC_Club/'
          element={
            <Testimonial
              name='Harry Potter'
              text='I had a great experience at PC Club!'
              img={require('../../assets/harry.jpg')}
            />
          }
        />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('Harry Potter')).toBeInTheDocument();
});
