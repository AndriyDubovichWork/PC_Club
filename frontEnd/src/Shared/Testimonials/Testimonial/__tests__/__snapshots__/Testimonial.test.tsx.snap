// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial match snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-er2eq5-MuiPaper-root"
>
  <div>
    <span
      className="Text"
    >
      "
      I had a great experience at PC Club! The staff was friendly and helpful. I highly recommend this store.
      "
    </span>
  </div>
  <div
    className="Author"
  >
    <img
      alt="testimonial"
      className="Img"
      loading="lazy"
      src="harry.jpg"
    />
    <h3>
      Harry Potter
    </h3>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19pa9qe-MuiSvgIcon-root"
      data-testid="FormatQuoteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
      />
    </svg>
  </div>
</div>
`;
